legacy/web/src/main/java/org/jboss/as/web/WebMigrateOperation.java

```diff
@@ -698,6 +709,55 @@ private void migrateAccessLog(Map<PathAddress, ModelNode> newAddOperations, Mode
         newAddOperations.put(newAddress, add);
     }
 
+    private String migrateAccessLogPattern(String legacyPattern) {
+        Matcher m = ACCESS_LOG_PATTERN.matcher(legacyPattern);
+        StringBuilder sb = new StringBuilder();
+        int lastIndex = 0;
+        while (m.find()) {
+            sb.append(legacyPattern.substring(lastIndex, m.start()));
+            lastIndex = m.end();
+            sb.append("%{");
+            sb.append(m.group(2));
+            sb.append(",");
+            sb.append(m.group(1));
+            sb.append("}");
+        }
+        sb.append(legacyPattern.substring(lastIndex));
+        return sb.toString();
+    }
+
+    private void migrateAccessLogValve(Map<PathAddress, ModelNode> newAddOperations, ModelNode newAddOp, PathAddress address, ModelNode legacyAddOps, List<String> warnings) {
+        ModelNode add = createAddOperation(VALVE_ACCESS_LOG_ADDRESS);
+        final ModelNode params = newAddOp.get(WebValveDefinition.PARAMS.getName());
+        //TODO: parse the pattern and modify to Undertow version
+        final ModelNode patternNode = params.get(Constants.PATTERN);
+        if(patternNode.isDefined()) {
+            add.get(Constants.PATTERN).set(migrateAccessLogPattern(patternNode.asString()));
+        }
+        add.get(Constants.PREFIX).set(params.get(Constants.PREFIX).clone());
+        add.get(Constants.SUFFIX).set(params.get(Constants.SUFFIX).clone());
+        add.get(Constants.ROTATE).set(params.get("rotatable").clone());
+        add.get(Constants.EXTENDED).set(newAddOp.get(Constants.EXTENDED).clone());
+        if(params.hasDefined(Constants.DIRECTORY)){
+            add.get(Constants.DIRECTORY).set(params.get(Constants.DIRECTORY).clone());
+        }
+        if(params.hasDefined("conditionIf")) {
+             add.get(Constants.PREDICATE).set("exists(%{r," + params.get("conditionIf").asString() + "})");
+        }
+        if(params.hasDefined("conditionUnless")) {
+            add.get(Constants.PREDICATE).set("not exists(%{r," + params.get("conditionUnless").asString() + "})");
+        }
+        if(params.hasDefined("condition")) {
+            add.get(Constants.PREDICATE).set("not exists(%{r," + params.get("conditionUnless").asString() + "})");
+        }
```

Comment:
Is this correct? Seems as copy paste error. Shoudln't there be `params.get("condition")`?